jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
> There is currently no SD attribute defined for ABCModelingEngine

That's true, but "by the specs" I'm refering to https://publicwiki-01.fraunhofer.de/SimPhoNy-Project/index.php/General_requirements
I think that document is a bit misleading, because a model is described by CM SD BC and SP, but Wrapper is described by CM BC SP and State-Data. Currently, as you wrote, ABCModellingEngine does not have an SD attribute, but what I understand from the document and from the discussion we had in Jyväskylä meeting, we're going to have it beside the other 3. Please, correct me if I'm wrong: my only intention here is to make JyuLB wrappers structurally close to other wrappers, or what they are going to be, once updated.